src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -811,6 +863,46 @@ private void checkParamTags(final List<JavadocTag> tags,
         }
     }
 
+    /**
+     * Checks the order for {@code param} tags.
+     * @param tags the tags to check
+     * @param allParams list of parameters
+     * @param allTypeParams list of type parameters
+     */
+    private void checkParamTagsOrder(final List<JavadocTag> tags,
+                                     final List<DetailAST> allParams,
+                                     final List<DetailAST> allTypeParams) {
+        // To store the actual position of the parameter, which will be
+        // later used for validating order of the params
+        int paramPosition = 0;
+        // Will the total no of type param present to validate order of parameters.
+        final int noOfTypeParams = allTypeParams.size();
+
+        // Loop over the tags, checking to see they exist in the params.
+        for (JavadocTag tag : tags) {
+            if (!tag.isParamTag()) {
+                continue;
+            }
+
+            final String arg1 = tag.getFirstArg();
+            int actualPosition = indexOfParam(allParams, arg1) + noOfTypeParams;
+
+            final boolean isTypeParam = CommonUtil.startsWithChar(arg1, '<')
+                    && CommonUtil.endsWithChar(arg1, '>');
+            if (isTypeParam) {
+                actualPosition = indexOfTypeParameter(allTypeParams,
+                        arg1.substring(1, arg1.length() - 1));
+            }
```

Comment:
why not add
````
} else {
  actual = indexOfParam(....)
}
````

Instead of executing it all the time?

Reply:
Yeah, I think this would be better. I'll make this change.

Reply:
Done